src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,330 @@
+end
+
+"""
+    struct BoundaryConditionViscousWall{V, H}
+
+Creates a wall-type boundary conditions for the compressible Navier-Stokes equations.
+The fields `boundary_condition_velocity` and `boundary_condition_heat_flux` are intended
+to be boundary condition types such as the `NoSlip` velocity boundary condition and the
+`Adiabatic` or `Isothermal` heat boundary condition.
+"""
+struct BoundaryConditionViscousWall{V, H}
+  boundary_condition_velocity::V
+  boundary_condition_heat_flux::H
+end
+
+"""
+    struct NoSlip{F}
+
+Creates a no-slip boundary condition with field `boundary_value_function`, which
+should be a function with signature `boundary_value_function(x, t, equations)`
+and should return a `SVector{NDIMS}` whose entries are the velocity vector at a
+point `x` and time `t`.
+"""
+struct NoSlip{F}
+  boundary_value_function::F # value of the velocity vector on the boundary
+end
+
+"""
+    struct Isothermal{F}
+
+Creates an isothermal temperature boundary condition with field `boundary_value_function`,
+which should be a function with signature `boundary_value_function(x, t, equations)` and
+return a scalar value for the temperature at point `x` and time `t`.
+"""
+struct Isothermal{F}
+  boundary_value_function::F # value of the temperature on the boundary
+end
+
+"""
+    struct Adiabatic{F}
+
+Creates an adiabatic temperature boundary condition with field `boundary_value_function`,
+which should be a function with signature `boundary_value_function(x, t, equations)` and
+return a scalar value for the normal heat flux at point `x` and time `t`.
+"""
+struct Adiabatic{F}
+  boundary_value_normal_flux_function::F # scaled heat flux 1/T * kappa * dT/dn
+end
+
+@inline function (boundary_condition::BoundaryConditionViscousWall{<:NoSlip, <:Adiabatic})(flux_inner, u_inner, normal::AbstractVector,
+                                                                                           x, t, operator_type::Gradient,
+                                                                                           equations::CompressibleNavierStokesEquations2D)
+  v1, v2 = boundary_condition.boundary_condition_velocity.boundary_value_function(x, t, equations)
+  return SVector(u_inner[1], v1, v2, u_inner[4])
+end
+
+@inline function (boundary_condition::BoundaryConditionViscousWall{<:NoSlip, <:Adiabatic})(flux_inner, u_inner, normal::AbstractVector,
+                                                                                           x, t, operator_type::Divergence,
+                                                                                           equations::CompressibleNavierStokesEquations2D)
+  # rho, v1, v2, _ = u_inner
+  normal_heat_flux = boundary_condition.boundary_condition_heat_flux.boundary_value_normal_flux_function(x, t, equations)
+  v1, v2 = boundary_condition.boundary_condition_velocity.boundary_value_function(x, t, equations)
+  tau_1n, tau_2n = flux_inner[2:3] # extract fluxes for 2nd and 3rd equations
```

Comment:
```suggestion
  _, tau_1n, tau_2n = flux_inner # extract fluxes for 2nd and 3rd equations
```
Does the current version allocate?

Reply:
I don't believe so, since `flux_inner` is an `SVector` for all current solvers. Is the suggested change guaranteed to not allocate?

Reply:
Nevermind - it does indeed allocate. Thanks for catching that! Fixing now